microprofile/WFLY-11529_subsystem_metrics.asciidoc

```diff
@@ -0,0 +1,145 @@
+= [WFLY-11529] Expose WildFly Metrics
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,metrics,openshift
+
+== Overview
+
+The goal of this feature is to improve observability of WildFly by exposing its own metrics to monitoring tools such as https://prometheus.io[Prometheus]
+and the https://docs.openshift.com/container-platform/3.11/architecture/infrastructure_components/web_console.html[OpenShift Console].
+
+This feature exposes the WildFly metrics (from the subsystem and deployment management models) to the `/metrics` HTTP endpoint in the Prometheus format.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11529[WFLY-11529]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1169[EAP7-1169]
+* https://issues.jboss.org/browse/EAP7-1026[EAP7-1026] - EAP7-1026 Expose EAP Metrics so that it can be consumed by Kiali (Istio) Console
+** in this feature, base metrics (e.g. JVM) and application metrics (from MicroProfile Metrics API) were exposed but there were none coming
+  from WildFly subsystems and deployments
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:rsvoboda@redhat.com[Rostislav Svoboda]
+
+=== Other Interested Projects
+
+* Prometheus MUST be able to consume and aggregate WildFly metrics.
+
+== Requirements
+
+* Expose WildFly metrics from subsystem and deployment mananagement model to the `/metrics` HTTP endpoint in the Prometheus format.
+* Provide mechanism to be _compatible_ with the WildFly metrics exposed by EAP images in OpenShift (that uses Prometheux JMX exporter as configured in https://github.com/jboss-container-images/jboss-eap-modules/blob/master/jboss/container/eap/prometheus/config/7.2/artifacts/opt/jboss/container/prometheus/etc/jmx-exporter-config.yaml[jmx-exporter-config.yaml])
+** compatibility means that the metric names and labels should be identical (or, if identity is not achievable, similar).
+
+=== Non-requirements
+
+* WildFly metrics are *NOT* exposed to the `/metrics` HTTP endpoint in the JSON format.
```

Comment:
I would prefer to have both Prometeus and JSON so everything under `/metrics` is consistent.

How complex would it be to have JSON enabled?